### DISCUSSION TITLE
How to Add JWT Token Support Globally in Scalar for a .NET 9 Application?

### DISCUSSION THREAD

**Discussion by recepgunes1** [scalar/scalar#4007] [Q&A] [answered]
I'm working on enabling Scalar in my .NET 9 application and need to configure it to use JWT tokens globally, similar to how Swagger allows adding JWT token authentication globally. However, I couldn't find any documentation or examples on how to achieve this with Scalar.

Is it possible to set up global JWT token support in Scalar? If so, how can this be implemented? Any examples or guidance would be greatly appreciated!

**Comment by IngoManthey** [comment]
@recepgunes1 
builder.Services.AddOpenApi(options =>
{
    options.AddDocumentTransformer<BearerSecuritySchemeTransformer>();
});

and 
app.MapScalarApiReference(options =>
{
    // Bearer
    options
        .WithPreferredScheme("Bearer") // Security scheme name from the OpenAPI document
        .WithHttpBearerAuthentication(bearer =>
        {
            bearer.Token = "your-bearer-token";
        });
    options
        .WithTitle("Customer Portal API")
        .WithDefaultHttpClient(ScalarTarget.CSharp, ScalarClient.HttpClient);
});

**Reply by xC0dex** [reply]
@IngoManthey thanks for the reply. In addition, [here](https://github.com/scalar/scalar/blob/main/documentation/integrations/dotnet.md#authentication) is the link to the docs.

**Reply by recepgunes1** [reply]
Thank you for the response, but I don't have a non-generic AddDocumentTransformer method available. Should I create a custom class to address this? Should I also add any NuGet package? If so, could you provide an example?

Code:
```
//other codes
services.AddOpenApi();
//other codes
app.MapOpenApi();
app.MapScalarApiReference(options =>
{
    options.WithTheme(ScalarTheme.Moon)
        .WithDarkMode(true)
        .WithDefaultHttpClient(ScalarTarget.CSharp, ScalarClient.HttpClient)
        .WithDarkModeToggle(false)
        .WithPreferredScheme("Bearer")
        .WithHttpBearerAuthentication(bearer =>
        {
            bearer.Token = "your-bearer-token";
        });
    options.Authentication = new ScalarAuthenticationOptions
    {
        PreferredSecurityScheme = "Bearer"
    };
});
```

Nuget packages:
- Microsoft.AspNetCore.OpenApi (9.0.0)
- Scalar.AspNetCore (1.2.45)

**Reply by xC0dex** [reply]
Hi @recepgunes1, were you able to resolve your problem?

**Reply by recepgunes1** [reply]
Hi @xC0dex, the issue remains unresolved despite trying every Scalar update. In version 1.2.56, there is an authentication combo box, but it appears to be empty.

![image](https://github.com/user-attachments/assets/540637a2-6719-42cd-b6c3-03a42e81a835)

**Reply by xC0dex** [reply]
Did you implement a custom document transformer which adds the security scheme to the OpenAPI document? [Here](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/openapi/customize-openapi?view=aspnetcore-9.0&source=recommendations#use-document-transformers) are the official Microsoft docs. I also built an extension library to make it a bit easier. [Here](https://github.com/xC0dex/APIWeaver/blob/main/docs/Getting-Started.md#authentication) are the docs.

**Reply by recepgunes1** [reply, marked as answer]
I installed `APIWeaver.OpenApi` and made some changes to the configuration options, which worked as expected. Below is the `AddOpenApi` registration code:

```
services.AddOpenApi(options =>
{
    options.AddSecurityScheme(JwtBearerDefaults.AuthenticationScheme, scheme =>
    {
        scheme.In = ParameterLocation.Header;
        scheme.Type = SecuritySchemeType.Http;
        scheme.Scheme = JwtBearerDefaults.AuthenticationScheme;
        scheme.BearerFormat = "JWT";
    });
    options.AddAuthResponse();
});
```
Thank you @xC0dex .

**Reply by hoangviet24** [reply]
Thank you my god

**Comment by seigodwin** [comment]
Thank you so much @xC0dex
